### PR TITLE
Making models

### PR DESCRIPTION
**Pull request by Dpalazzari**
In this PR:

- Creates User Model
  - Adds User model relationships
  - Adds User model relationships tests

- Creates Adjective model
  - Adds adjective model relationships
  - Adds adjective model relationship tests

- Creates assignment model
  - Adds assignment model relationships
  - adds assignment model relationships tests

- Creates description model
  - Adds description model relationships
  - Adds description model relationships tests

Worked on the branch with @wlffann 

Our database schema is one of a kind, so we need @kheppenstall @lucyconklin @akintner to closely read, and make sure you understand the relationships. 

Feedback/requests for changes are expected.

FYI @case-eee

### CODE REVIEW

**Comment by akintner** [comment]
I like the way you handled all of the relationships on the user model. It all makes perfect sense, but I'm glad you left the comments in there. Nice work.

File: spec/factories/users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :user do
+    name Faker::GameOfThrones.character
+  end
```
Comment (lucyconklin) [inline spec/factories/users.rb]: Winter is coming.

File: app/models/assignment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Assignment < ApplicationRecord
+  # User assigned to do the describing
+  belongs_to :assignee, class_name: "User"
+  # User assigned to be desribed
+  belongs_to :assigned, class_name: "User"
+
+  enum status: %w(open closed)
+end
```
Comment (lucyconklin) [inline app/models/assignment.rb]: re: open/closed I imagine this could be a way to find users who did not complete the exercise yet.

**Comment by lucyconklin** [comment]
My only question is: Should descriptions and assignments be linked? By, say, another column on the assignments table for description? Or will we handle this by changing the status to "closed" when a description is made matching that assignment?

File: spec/factories/users.rb (hunk 1 above)
Comment (Dpalazzari) [inline spec/factories/users.rb]: A lion doesn't concern itself with the opinion of sheep.

File: app/models/assignment.rb (hunk 2 above)
Comment (Dpalazzari) [inline app/models/assignment.rb]: correct!

**Comment by Dpalazzari** [comment]
@lucyconklin I think we will handle it by changing the enum after the assigned assignments have been submitted. Does that make sense?

File: spec/factories/users.rb (hunk 1 above)
Comment (akintner) [inline spec/factories/users.rb]: ❄️

File: app/models/adjective.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Adjective < ApplicationRecord
+  has_many :descriptions
+  has_many :users, through: :descriptions, source: :describee
```
Comment (kheppenstall) [inline app/models/adjective.rb]: @Dpalazzari @wlffann What is the purpose of this has many relationship? Do we have any need to go from an adjective to all users that have been described with that adjective right now?

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  # users that this user has described
+  has_many :described_users, through: :described, source: :describer
+  
+  # who this user has been assigned to describe
+  has_many :assignments, foreign_key: :assignee_id
+  has_many :users_to_describe, through: :assignments, source: :assigned
+  # who has been assigned to describe this user
+  has_many :assigned_to, foreign_key: :assigned_id, class_name: "Assignment"
+  has_many :users_described_by, through: :assigned_to, source: :assignee
+end
```
Comment (kheppenstall) [inline app/models/user.rb]: 👍 On this page. Love the descriptive names! Why do we need lines 17-19? Just trying to think of the case of starting with a user and needing to find all the people that have been assigned to describe him / her.

**Comment by kheppenstall** [comment]
👍 Looks great to me! I left a couple comments wondering about why we created some relationships. They are not really important though and I'm happy for this to be merged in whenever. Good stuff!

File: app/models/adjective.rb (hunk 3 above)
Comment (Dpalazzari) [inline app/models/adjective.rb]: Are you referring to the has_many :users @kheppenstall ? I doubt we ever would use it, to be honest. I think we just tried to implement every relationship we could see on the database. Easy to remove though. What does @wlffann think?

File: app/models/user.rb (hunk 4 above)
Comment (Dpalazzari) [inline app/models/user.rb]: I'll get back to you on this comment. Let me think! @kheppenstall

Comment (Dpalazzari) [inline app/models/user.rb]: Perhaps we might implement some functionality that will show all the users who have judged you; later on in the app.

**Pull request merged**